src/test/java/org/opensearch/security/dlic/dlsfls/StarTreeDlsFlsTest.java

```diff
@@ -0,0 +1,476 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.security.dlic.dlsfls;
+
+import org.apache.http.HttpStatus;
+import org.junit.Assert;
+import org.junit.Test;
+
+import org.opensearch.action.admin.indices.create.CreateIndexRequest;
+import org.opensearch.action.admin.indices.forcemerge.ForceMergeRequest;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.action.support.WriteRequest.RefreshPolicy;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.security.test.DynamicSecurityConfig;
+import org.opensearch.security.test.helper.rest.RestHelper.HttpResponse;
+import org.opensearch.transport.client.Client;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+/**
+ * Integration tests for star tree functionality with DLS/FLS restrictions.
+ * Tests that star tree queries are disabled when DLS/FLS is applied.
+ */
+public class StarTreeDlsFlsTest extends AbstractDlsFlsTest {
+
+    private static final String STARTREE_INDEX_NAME = "startree_sales";
+    private static final String REGULAR_INDEX_NAME = "regular_sales";
+
+    protected void setupStarTreeTest() throws Exception {
+        final Settings settings = Settings.EMPTY;
+        final DynamicSecurityConfig dynamicSecurityConfig = new DynamicSecurityConfig().setSecurityRoles("roles_startree.yml")
+            .setSecurityRolesMapping("roles_mapping_startree.yml")
+            .setSecurityInternalUsers("internal_users_startree.yml");
+
+        setup(settings, dynamicSecurityConfig);
+    }
+
+    @Override
+    protected void populateData(Client tc) {
+        // Create star tree index with composite mapping
+        try {
+            tc.admin()
+                .indices()
+                .create(
+                    new CreateIndexRequest(STARTREE_INDEX_NAME).settings(
+                        Settings.builder()
+                            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+                            .put("index.composite_index", true)
+                            .put("index.append_only.enabled", true)
+                            .put("index.search.star_tree_index.enabled", true)
+                            .build()
+                    )
+                        .mapping(
+                            XContentFactory.jsonBuilder()
```

Comment:
nit: any chance we can convert this to raw json and use a text block for readability?